src/Microsoft.Identity.Web/WebAppExtensions/WebAppAuthenticationBuilderExtensions.cs

```diff
@@ -136,12 +136,19 @@ public static partial class WebAppAuthenticationBuilderExtensions
                     if (!string.IsNullOrWhiteSpace(login))
                     {
                         context.ProtocolMessage.LoginHint = login;
-                        context.ProtocolMessage.DomainHint = context.Properties.GetParameter<string>(
-                            OpenIdConnectParameterNames.DomainHint);
 
-                        // delete the login_hint and domainHint from the Properties when we are done otherwise
+                        // delete the login_hint from the Properties when we are done otherwise
                         // it will take up extra space in the cookie.
                         context.Properties.Parameters.Remove(OpenIdConnectParameterNames.LoginHint);
+                    }
+
+                    var domainHint = context.Properties.GetParameter<string>(OpenIdConnectParameterNames.DomainHint);
+                    if (!string.IsNullOrWhiteSpace(domainHint))
+                    {
+                        context.ProtocolMessage.DomainHint = domainHint;
+
+                        // delete the domainHint from the Properties when we are done otherwise
+                        // it will take up extra space in the cookie.
                         context.Properties.Parameters.Remove(OpenIdConnectParameterNames.DomainHint);
```

Comment:
Thanks for the contribution, we already have a [test verifying the doman_hint is set](https://github.com/AzureAD/microsoft-identity-web/blob/master/tests/Microsoft.Identity.Web.Test/WebAppExtensionsTests.cs#L406), but not sure if we want to check one or the other as well, or this test is enough...personally, I'm okay with this current test and the work done here.